### DISCUSSION TITLE
Is there a way to assert the number of checks done?

### DISCUSSION THREAD

**Discussion by thejohnfreeman** [onqtam/doctest#769] [Q&A]
I'm trying to test some async code that uses callbacks. I have `CHECK`s in my callback. I don't want my test to trivially pass because the callback was never called. At the end of my test, I want to assert that a specific or minimum number of checks were done. Is that possible in the current framework, or is this a feature request? I'll open an issue if it's a request. Thank you!

**Comment by onqtam** [comment]
Perhaps you could hack your way around and expose `CurrentTestCaseStats` / `numAssertsCurrentTest`:
https://github.com/doctest/doctest/blob/master/doctest/doctest.h#L2017

Perhaps you could hack it similarly to this: https://github.com/doctest/doctest/issues/345#issuecomment-593625010

**Comment by mitchgrout** [comment]
I have had experience with similar scenarios in which the testable component is within a callback which is called N many times. The preferred route I had was to instead aggregate results into a collection, and perform the assertions on _that_ instead. That way, if the callback is inadvertently run or less than expected, you can check this using a pretty standard mechanism.

**Reply by mitchgrout** [reply]
As a secondary option, you could consider using a mock/spy-type callback that is able to record information about calls, and assert on that instead.
